### PR TITLE
Skip the host header in the grpc metadata

### PR DESCRIPTION
**Pull request by lei-tang**
gRPC requests (HTTP2) with a host header will lead some gRPC servers to reject it, so the host header is skipped in the grpc metadata.

### CODE REVIEW

**Comment by lei-tang** [comment]
https://github.com/cloudendpoints/esp/issues/423

**Comment by lizan** [comment]
tests?

**Comment by lei-tang** [comment]
Working on the test, will update once done.

**Comment by lei-tang** [comment]
@lizan , @qiwzhang : a test has been added. Please take a look.

File: test/transcoding/bookstore-server.cc [hunk 1]
```diff
@@ -99,22 +114,28 @@ class BookstoreServiceImpl : public Bookstore::Service {
 
   // Bookstore::Service implementation
 
-  ::grpc::Status ListShelves(::grpc::ServerContext*,
+  ::grpc::Status ListShelves(::grpc::ServerContext* ctx,
                              const ::google::protobuf::Empty* request,
                              ListShelvesResponse* reply) {
     std::cerr << "GRPC-BACKEND: ListShelves" << std::endl;
     PrintRequest(*request);
+    if (printmetadata_) {
```
Comment (qiwzhang) [inline test/transcoding/bookstore-server.cc]: Just a idea.   you can pass ctx to PrintRequest, change PrintRequest to also print metadata which is controlled by the global flag.  This is much cleaner.

Comment (lei-tang) [reply test/transcoding/bookstore-server.cc]: Done.

**Pull request merged**